### DISCUSSION TITLE
How to work with "Projects"

### DISCUSSION THREAD

**Discussion by Gerschtli** [nix-community/nix-on-droid#160] [Ideas]
Starting a discussion instead of adding unrelated comments to https://github.com/t184256/nix-on-droid/pull/147 :)

I like the new Project board but was wondering where to put potential ideas if they are not necessarily part of the 21.11 roadmap..

Maybe we open one additional big board without any release connection or we open separate topic-specific boards. I think for our current man power, two boards (release-specific and one for the rest) is sufficient.

Just to dump a few ideas I had in mind (order does not show priority), so we are faster agreeing or disagreeing on ideas and topics and its priority for the project:

* flakify the building process
* as good as possible nixify (or flakify) the test setup
* regular building of packages in CI we provide overlays for, to remove the nixpkgs pin in there (https://github.com/t184256/nix-on-droid/issues/52)
* is service management (supervisord or sth else) possible? manual start/stop or even on boot starting? See https://github.com/t184256/nix-on-droid/issues/54
* render options as html/man page (see https://github.com/t184256/nix-on-droid/issues/36)
* a nixos-option equivalent for nix-on-droid
* remove necessity of hardcoded store paths for proot (to remove --impure flag for flakes)
* support some termux<->android stuff, that is somtimes requested in issues (https://github.com/t184256/nix-on-droid/issues/133, https://github.com/t184256/nix-on-droid/issues/118)
* more overlays via patches from termux-packages? (https://github.com/t184256/nix-on-droid/issues/78)
* bootstrap device with flake setup without any channels (e.g. via prompt at initial start: flake or channel?)

**Comment by t184256** [comment]
Sounds good. I've imagined some middle-ground, namely "flakify everything", "on-device proot building" or "service management", created after there's some plan to move forward with these. I also like that issues can belong to multiple boards at once. So, go ahead and create what you like

* "flakify everything" and "documentation" sound actionable and decomposable.
* "on-device proot building" doesn't. I remember last time non-prebuilt toolchain was substantially broken and I haven't seen much movement on it since, so I'm waiting this out. Wonder if `--impure` is somehow avoidable without it with fixed-output derivations.
* "more overlays" doesn't sound like a finite continuous effort, so I'd rather not.
* "service management" sounds fun up until I think about integrating it into Android app code and then it doesn't
* "termux<->android stuff" similarly scares me away
* a nixos-option equivalent for nix-on-droid --- what's that?
* regular building of packages in CI we provide overlays for, to remove the nixpkgs pin in there: sounds one-off

**Reply by Gerschtli** [reply]
So how about creating "flakify everything" (maybe include research for pure builds), "service management" and "documentation + discoverability" projects. The latter one would also include stuff like nixos-option thing (a CLI to read the current value of a set config option, home-manager has something like that, too).

For topics like proot on-device building, somehow porting termux overlays and termux<->android integration we could just create separate discussions to gather ideas and current experiment results.

For the CI: Yeah we should just write that workflow yaml, then we are done :D
